surefire-providers/surefire-junit-platform/src/test/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProviderTest.java

```diff
@@ -891,6 +893,57 @@ public void parsesConfigurationParameters()
         assertEquals( "EOF", provider.getConfigurationParameters().get( "qux" ) );
     }
 
+    @Test
+    public void shouldFilterTestMethod()
+    {
+        ProviderParameters providerParameters = providerParametersMock();
+        TestListResolver testListResolver = new TestListResolver( "**/*Test#test*" );
+        assertFalse( testListResolver.isEmpty() );
+        assertFalse( testListResolver.isWildcard() );
+        TestRequest request = new TestRequest( null, null, testListResolver, 0 );
+        when( providerParameters.getTestRequest() ).thenReturn( request );
+
+        JUnitPlatformProvider provider = new JUnitPlatformProvider( providerParameters );
+
+        assertEquals( provider.getFilters().length, 1 );
+
+        assertThat( provider.getFilters()[0] )
+            .isInstanceOf( TestMethodFilter.class );
+
+        assertThat( (TestListResolver) getInternalState( provider.getFilters()[0], "testListResolver" ) )
+            .isSameAs( testListResolver );
+    }
+
+    @Test
+    public void shouldNotFilterEmpty()
+    {
+        ProviderParameters providerParameters = providerParametersMock();
+        TestListResolver testListResolver = new TestListResolver( "" );
+        assertTrue( testListResolver.isEmpty() );
+        assertFalse( testListResolver.isWildcard() );
+        TestRequest request = new TestRequest( null, null, testListResolver, 0 );
+        when( providerParameters.getTestRequest() ).thenReturn( request );
+
+        JUnitPlatformProvider provider = new JUnitPlatformProvider( providerParameters );
+
+        assertEquals( provider.getFilters().length, 0 );
+    }
+
+    @Test
+    public void shouldNotFilterWildcard()
+    {
+        ProviderParameters providerParameters = providerParametersMock();
+        TestListResolver testListResolver = new TestListResolver( "*.java" );
+        assertTrue( testListResolver.isWildcard() );
```

Comment:
ok  `*.java` is wildcard ... 
but `**/*.java` should be wildcard or not?

Reply:
They are both the same in this case.
The TLR assumes wildcard path if not specified.

Reply:
what result will be of?
```
new TestListResolver( '**/*.java' ). isWildcard()
```